.circleci/config.yml

```diff
@@ -0,0 +1,100 @@
+version: 2
+jobs:
+  relay:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps: 
+      - checkout
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn relay
+  lint:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps: 
+      - checkout
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn lint 
+  type-check:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps: 
+      - checkout
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn type-check
+  deploy:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps:
+      - checkout
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn semantic-release
+      - run: cp .env.oss .env
+      - run: yarn deploy-storybook
+  test:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps: 
+      - checkout
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn test -w 4
+  update-cache:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps:
+    - checkout
+    - restore_cache:
+        keys:
+        - yarn-deps-{{ checksum "yarn.lock" }}
+        - yarn-deps-
+    - run: yarn --check-files
+    - save_cache:
+        key: yarn-deps-{{ checksum "yarn.lock" }}
+        paths:
+        - ./node_modules
```

Comment:
This task is interesting... its sole purpose is to ensure that the cache gets populated. Because this job runs in parallel to all the other jobs we have a worse case performance scenario where all jobs end up installing their dependencies. This is the only job that stories them in CI though. The cache persists across multiple workflows and is immutable, so the worse case can ever only happen once. By the nature of CircleCI's cache fallback mechanism, we should *never* hit the worse case (unless they remove the cache out from under us). The other outlier scenario is when a dependency updates. That'll trigger a new cache write, but all the jobs will have to install those missing dependencies on that first run. All subsequent runs should be 🙆‍♂️ .